src/evict/evict_page.c

```diff
@@ -195,25 +195,10 @@ __evict_page_dirty_update(WT_SESSION_IMPL *session, WT_REF *ref, int exclusive)
 		break;
 	case WT_PM_REC_MULTIBLOCK:			/* Multiple blocks */
 		/*
-		 * There are two cases in this code.
-		 *
-		 * First, an in-memory page that got too large, we forcibly
-		 * evicted it, and there wasn't anything to write. (Imagine two
-		 * threads updating a small set keys on a leaf page. The page is
-		 * too large so we try to evict it, but after reconciliation
-		 * there's only a small amount of data (so it's a single page we
-		 * can't split), and because there are two threads, there's some
-		 * data we can't write (so we can't evict it). In that case, we
-		 * take advantage of the fact we have exclusive access to the
-		 * page and rewrite it in memory.)
-		 *
-		 * Second, a real split where we reconciled a page and it turned
-		 * into a lot of pages.
+		 * A real split where we reconciled a page and it turned into a
+		 * lot of pages.
 		 */
-		if (mod->mod_multi_entries == 1)
-			WT_RET(__wt_split_rewrite(session, ref));
-		else
-			WT_RET(__wt_split_multi(session, ref, exclusive));
+		WT_RET(__wt_split_multi(session, ref, exclusive));
```

Comment:
I don't believe we will ever get here (in the branch) with a single page and `WT_PM_REC_MULTIBLOCK` set.  We check how many pages were created during reconciliation, and set `EMPTY` for zero, `REPLACE / REWRITE` for one and `MULTIBLOCK` for more than one.

Even if we did somehow end up here, we would have come via `__rec_split_col/row` so everything would be set up as expected by `__rec_col/row_merge`.